src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -382,23 +383,29 @@ UCS_PROFILE_FUNC_ALWAYS(ucs_status_t, uct_ib_mlx5_devx_reg_atomic_key,
     uct_ib_mr_type_t mr_type  = uct_ib_md_get_atomic_mr_type(&md->super);
     uct_ib_mlx5_devx_mr_t *mr = &memh->mrs[mr_type];
     uint8_t mr_id             = uct_ib_md_get_atomic_mr_id(&md->super);
+    uint32_t index            = 0;
     ucs_status_t status;
     int is_atomic;
 
+    if (memh->smkey_mr != NULL) {
+        index = md->super.mkey_by_name_reserve.size / 2 +
```

Comment:
what is this calc? pls add comment

Reply:
first half of reserve.size is for direct key allocation, second half is to mirror offset/index obtained for atomic rkey.

Reply:
added function, using it twice.